### ISSUE TITLE
Compiler author type

### ISSUE THREAD

**Issue by adam3smith** [citation-style-language/zotero-bits#43]
Compiler - as of an anthology or the collected works of XY - is a separate author type with it's own label - in CMoS it's comp. or comps.
This is mentioned in CMoS 14.88
http://www.chicagomanualofstyle.org/16/ch14/ch14_sec088.html

<blockquote>The edited, compiled, or translated work of one author is normally listed with the author’s name appearing first and the name(s) of the editor(s), compiler(s), or translator(s) appearing after the title, preceded by edited by or ed., compiled by or comp., or translated by or trans.</blockquote>

The poster here http://forums.zotero.org/discussion/19095/needs-expanded-author-types
also points to CMoS 15.96 (which doesn't seem to exist online) and provides this sample citation:

Manuel Santos, comp., <i>The Collected Works of Henritta Kahn</i> (Boston: I. J. Filbert, 1989).

I'll note that, while clearly extant, it's also exceedingly rare - there is, in fact, not a single example using comp/comps in either the 15th or the 16th edition of CMoS online.

**Comment by ACProctor** [comment]
I would certainly recommending adding a role of 'compiler'. FHISO (http://fhiso.org) are currently looking to produce a data vocabulary for citation-elements, and many of their issues with the very broad range of sources cited in genealogy can be found on their mailing lists. One of the things that was immediately obvious for a simple book citation was the absence of 'compiler'.

**Comment by bwiernik** [comment]
What is the relationship between a "compiler" and the French role "directeur du publication" (CSL: `editorial-director`)?

**Comment by bar-foo-x** [comment]
There's not really much connection with directeur du publication. Any work published in France must have a directeur du publication, who has a particular legal role as responsible for the publication. Outside of France, and maybe other countries with similar legal frameworks, there's no equivalent.

Only certain works have compilers, however. A typical case is anthologies, where the compiler selects and orders the material without substantially modifying it. (This is somewhat different from the usage for genealogies.)

Compiler seems to be the **only** type of book creator in CMoS that is not accounted for by Zotero: 14.103 (17th ed.) says: "In full note citations and in bibliographies, the abbreviation _ed._ or _eds._, _comp._ or _comps._, or _trans._ follows the name, preceded by a comma."  The example they give is:

Harold Schechter and Kurt Brown, comps., _Killer Verse: Poems of Murder and Mayhem_ (London: Everyman Paperback Classics, 2011), 33.

MLA also calls for "comp." in bibliographies. Journal style guides that follow Chicago explicitly call for "comp." / "compiled by" (e.g., [https://www.tandf.co.uk/journals/authors/style/reference/tf_F.pdf]).

It's a real annoyance that CSL/Zotero doesn't handle this (admittedly, I say this as an outlier, someone who wrote a book about anthologies). "Editor" is not an accurate description of the compiler's role, so to be correct it's necessary to go through and manually edit the final document.

**Comment by denismaier** [comment]
@bwiernik What do you say? That will be covered by your new name data model as proposed [here](https://github.com/citation-style-language/csl-evolution/issues/29), right?

If yes, do you think there's a need for a distinct role compiler? Is that common enough?

**Comment by bwiernik** [comment]
I think "compiler" is distinct enough to get its own variable. It's not just a type-specific form of author I don't think.

**Comment by denismaier** [comment]
@bwiernik As you address this in #202, perhaps you can add the Zotero label.

**Comment by bwiernik** [comment]
Yes, that's the idea, but there aren't keywords for changing tags automatically.
